### PR TITLE
fix(ScyllaYaml): experimental_feature as list

### PR DESCRIPTION
**Pull request by aleksbykov**
experimental_feature should be a list in scylla.yaml instead
of dict

https://docs.scylladb.com/operating-scylla/scylla-yaml/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/provision/scylla_yaml/scylla_yaml.py [hunk 1]
```diff
@@ -252,7 +252,7 @@ def set_authorizer(cls, authorizer: str):
     skip_wait_for_gossip_to_settle: int = -1
     force_gossip_generation: int = -1
     experimental: bool = False
-    experimental_features: dict = {}
+    experimental_features: list = []
```
Comment (dkropachev) [inline sdcm/provision/scylla_yaml/scylla_yaml.py:255]:
```suggestion
    experimental_features: list[str] = []
```

**Review by dkropachev — CHANGES_REQUESTED** [review]

**Review by dkropachev — CHANGES_REQUESTED** [review]
Alex, could also please fix unit tests

**Review by juliayakovlev — DISMISSED** [review]
ok, after addressing of @dkropachev comment

**Review by enaydanov — DISMISSED** [review]

**Comment by roydahan** [comment]
@aleksbykov is this PR still needed?

**Comment by aleksbykov** [comment]
@roydahan yes, it is needed. Need approval from @dkropachev

**Review by dkropachev — APPROVED** [review]

**Review by vponomaryov — APPROVED** [review]

**Pull request merged**